ArjunaJTA/jta/classes/com/arjuna/ats/jta/common/JTAEnvironmentBean.java

```diff
@@ -586,30 +585,6 @@ public void setXaResourceOrphanFilters(List<XAResourceOrphanFilter> xaResourceOr
         }
     }
 
-    /**
-     * Returns if connections associated to XAResources that fail during prepare should be cleaned up immediately.
-     * TODO move to JDBC module as it's only for our own connection manager?
-     *
-     * Default: false.
-     * Equivalent deprecated property: com.arjuna.ats.jta.xaRollbackOptimization
-     *
-     * @return true for cleanup during prepare, false for cleanup during phase two rollback.
-     */
-    public boolean isXaRollbackOptimization()
```

Comment:
Public API change implies Narayana 8 should be discussed. Maybe we want to consider if some of this should be in a separate commit?

Reply:
BTW the Javadoc gives a clue actually as to why this was here it seems something for transactional driver perhaps

Reply:
> Public API change implies Narayana 8 should be discussed. Maybe we want to consider if some of this should be in a separate commit?

I see your point. On the other hand, `isXaRollbackOptimization` hasn't been working for a long while and nobody complained...so it is likely that nobody is using this configuration. @mmusgrov do you think we can include this change in a minor or (as Tom suggested) should we create a new JBTM for removing it and wait the next major?

Reply:
If the functionality is proven not to be working, could we consider to remove the "implementation" and log errors if someone tries to use the public API parts (setter and getter)? I am wondering now, does this pull request somehow prove the feature is not working?

Reply:
Nope, I didn't develop any test for that configuration property as I planned to remove it. Nevertheless, it should be possible to test it using the two tests I introduced with this PR (when I was confirming Wladimir's [analysis](https://groups.google.com/g/narayana-users/c/o6L7uMFmkvo), I used the test in `XAResourceRecordUnitTest` to check if the rollback optimization was working)

Reply:
@jmfinelli Just mark it deprecated (and fill in the release note field on the JIRA) and since the default is false there should be no difference between removing it and not removing it. Then if someone is relying on it to remove the connection there won't be any surprises for them.

I would *not* spend any time writing tests for a piece of broken functionality which, even if it did work, is no needed anyway.

Reply:
I'll do that. Thanks @mmusgrov

Reply:
I added `JTAEnvironmentBean:xaRollbackOptimization is deprecated with the intention of removing it in a future major release` to `Release Note Text`. Should I also modify `Release Note Status`?

Reply:
I would expect it's worth pointing out that invoking the function is no longer expected to have any affect on the behaviour of Narayana at runtime (IOW it's basically a no-op now).

Reply:
Release Note Text updated.